dotCMS/src/main/java/com/dotmarketing/portlets/cmsmaintenance/factories/CMSMaintenanceFactory.java

```diff
@@ -17,138 +18,175 @@
+ * <li>Templates.</li>
+ * <li>Links.</li>
+ * <li>Workflow History.</li>
+ * </ul>
  *
+ * @author root
+ * @since Mar 22, 2012
  */
 public class CMSMaintenanceFactory {
 
-	public static int deleteOldAssetVersions (Date assetsOlderThan) {
-		int counter = 0;
-		int auxCount = 0;
-
+    /**
+     * Deletes dotCMS object versions that are older than the specified date. Such a
+     * date is compared to the date in which a version of an object was created. The
+     * list of objects may include:
+     * <ul>
+     * <li>Contentlets - including HTML Pages as Content, Files as Content, Sites,
+     * etc.</li>
+     * <li>Containers.</li>
+     * <li>Templates.</li>
+     * <li>Links.</li>
+     * <li>Workflow History.</li>
+     * </ul>
+     *
+     * @param assetsOlderThan
+     *            The user-specified date up to which object versions will be
+     *            deleted.
+     * @return The total number of versions that were deleted.
+     */
+    public static int deleteOldAssetVersions(final Date assetsOlderThan) {
+        Logger.info(CMSMaintenanceFactory.class, "------------------------------------------------------------");
+        Logger.info(CMSMaintenanceFactory.class, " Executing the Drop Old Assets Versions tool");
+        Calendar startDate = Calendar.getInstance();
+        startDate.setTime(assetsOlderThan);
+        startDate.add(Calendar.YEAR, -2);
+        startDate.set(Calendar.HOUR_OF_DAY, 0);
+        startDate.set(Calendar.MINUTE, 0);
+        startDate.set(Calendar.SECOND, 0);
+        startDate.set(Calendar.MILLISECOND, 0);
+        final Calendar oldestInodeDate = getOldestInodeDate();
+        startDate = (null != oldestInodeDate ? oldestInodeDate : startDate);
+        Logger.info(CMSMaintenanceFactory.class, String.format("- Deleting versions older than '%s'",
+                UtilMethods.dateToHTMLDate(assetsOlderThan, "yyyy-MM-dd")));
+        Logger.info(CMSMaintenanceFactory.class, " ");
+        int totalRecords = 0;
+        while (startDate.getTime().before(assetsOlderThan) || startDate.getTime().equals(assetsOlderThan)) {
+            try {
+                HibernateUtil.startTransaction();
+                int deletedRecords = removeOldVersions(startDate);
+                // Run the drop tasks iteratively, moving forward in time
+                // DROP_OLD_ASSET_ITERATE_BY_SECONDS controls how many seconds to move forward
+                // in time for each iteration - default is to iterate by 30 days
+                startDate.add(Calendar.SECOND,
+                        Config.getIntProperty("DROP_OLD_ASSET_ITERATE_BY_SECONDS", 60 * 60 * 24 * 30));
+                // We should never go past the date the user entered
+                if (startDate.getTime().after(assetsOlderThan)) {
+                    break;
+                }
+                totalRecords += deletedRecords;
+                HibernateUtil.commitTransaction();
+            } catch (final Exception ex) {

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5534906)